src/listener_message.rs

```diff
@@ -15,18 +15,11 @@
 // Please review the Licences for the specific language governing permissions and limitations
 // relating to use of the SAFE Network Software.
 
-use sodiumoxide::crypto::sign::PublicKey;
 use socket_addr::SocketAddr;
 
+pub const REQUEST_MAGIC_CONSTANT: [u8; 4] = ['E' as u8, 'C' as u8, 'H' as u8, 'O' as u8];
```

Comment:
What I mean is that this makes it receive a single message type which is an array, or with an enum receive a multitude of message types and parse to the correct type. The enum wraps a bunch of the types which allows us to have more message types per connection. Otherwise we try and parse to type1 then type2 etc. if you see what I mean. Also the enum allows us to implement a the methods we wish to do on send/rec messages.